README.md

```diff
@@ -2,73 +2,126 @@
-  -i <value> | --iterations <value>
-        the number of iterations to run
-  --help
-        prints this usage text
-        
-$ bin/run --benchmark DatasetPerformance
+You may get Spark from http://github.com/apache/spark.
+
+Suppose the spark directory is ~/spark. To build:
+
+```
+cd ~/spark
+build/mvn -DskipTests -T 8C -Dskip -Phive -Phive-thriftserver package
 ```
 
-# TPC-DS
+In the above command, `-T 8C` means to use 8 threads to build; `-Phive -Phive-thriftserver` means to build with Hive support.
 
-## How to use it
-The rest of document will use TPC-DS benchmark as an example. We will add contents to explain how to use other benchmarks add the support of a new benchmark dataset in future.
+## Start a spark-shell with access to spark-sql-perf classes
 
-### Setup a benchmark
-Before running any query, a dataset needs to be setup by creating a `Benchmark` object. Generating
-the TPCDS data requires dsdgen built and available on the machines. We have a fork of dsdgen that
-you will need. It can be found [here](https://github.com/davies/tpcds-kit).  
+Supose the previous step that builds spark-sql-perf generated a folder scala-2.11 in ~/spark-sql-perf/target.
+
+```
+cd ~/spark
+bin/spark-shell --driver-class-path ~/spark-sql-perf/target/scala-2.11/classes
+```
+
+## Generate data, create tables, and create a TPCDS object
+
+Run the following commands in the spark-shell.
 
 ```
 import com.databricks.spark.sql.perf.tpcds.Tables
-// Tables in TPC-DS benchmark used by experiments.
-// dsdgenDir is the location of dsdgen tool installed in your machines.
-val tables = new Tables(sqlContext, dsdgenDir, scaleFactor)
-// Generate data.
-tables.genData(location, format, overwrite, partitionTables, useDoubleForDecimal, clusterByPartitionColumns, filterOutNullPartitionValues)
-// Create metastore tables in a specified database for your data.
-// Once tables are created, the current database will be switched to the specified database.
-tables.createExternalTables(location, format, databaseName, overwrite)
-// Or, if you want to create temporary tables
-tables.createTemporaryTables(location, format)
-// Setup TPC-DS experiment
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.SQLContext
+
+val sqlContext = SQLContext.getOrCreate(sc)
+
+// Create a Tables object without generating any data yet.
+// Record the locatin of dsdgen, and scale factor 1.
+val tables = new Tables(sqlContext, "~/tpc_ds/v2.4.0/tools", 1)
+
+// Call dsdgen to create .dat files (if not exists already) in ~/tpc_ds/v2.4.0/tools.
+// Then load into parquet files to be stored at "~/datafiles".
+// The five boolean parameters are:
+// - true  = overwrite
+// - false = partitionTables
+// - true  = useDoubleForDecimal
+// - false = clusterByPartitionColumns
+// - false = filterOutNullPartitionValues
+// See the source code src/main/scala/com/databricks/spark/sql/perf/tpcds/Tables.scala.
+tables.genData("~/datafiles", "parquet", true, false, true, false, false)
```

Comment:
needs update after #105
also, it was more readable to use descriptive val names than a "true, false, true, false, false" and a comment description.